<!-- -->
<launch>

	<arg name="base_control_port" default="/dev/U4"/>   <!--底层can分析仪-->
	<arg name="gps_port" default="/dev/ttyS1"/>		    <!-- GPS-->
	<arg name="stm32_port" default="/dev/U6"/>          <!-- STM32-->
	<arg name="max_speed"  default="30.0"/>
	<arg name="forsight_dis" default="5.0"/>
	<arg name="path_file" default="path"/>
	
<!--底层 -->
	<node name="base_control" pkg="little_ant" type="base_control" output="screen">
		<param name ="obd_can_port_name" value="$(arg base_control_port)"/>
		<param name ="stm32_port_name" value="$(arg stm32_port)"/>
		<param name ="stm32_baudrate" value="115200"/>
		<param name ="max_steering_speed" value="2.0"/> <!--前后帧转角最大差值 -->
	</node>
	
	<node pkg="novatel" type="novatel_node" name="novatel_location_node"  >
		<param name="port" value="$(arg gps_port)" />
		<param name="baudrate" value="115200" />
		<param name="log_commands" value="inspvaxb ontime 0.05" />
		<param name="is_lltoutm"   value="true"/>
		<param name="ll2utm_topic" value="/ll2utm_odom"/>
		<param name="inspvax_topic" value = "/gps" />
	</node>
	
	<node name="pid_tracking_node" pkg="path_tracking" type="pid_tracking_node" output="screen">
		<param name="utm_odom_topic" value="/ll2utm_odom"/>
		<param name="tracking_info_topic" value="/tracking_state"/>
		<param name="path_points_file" value="$(find recorder)/data/path_data/raw/$(arg path_file).txt"/>
		<param name="speed" value="$(arg max_speed)"/>						<!-- 跟踪速度  -->
		<param name="disThreshold" value="$(arg forsight_dis)"/>
		<param name="kp" value="-0.5"/>
	</node>
	
	<node  type="decision_making_node" pkg ="decision_making" name="decision_making_node" >  
		<param name ="traffic_mark_topic" value="/traffic_mark" />
	</node>
	
	
	<!-- record msgs -->
	
	<node pkg="novatel" type="novatel_node" name="novatelImu_node" output="screen">
		<param name="port" value="/dev/ttyS2" />
		<param name="baudrate" value="115200" />
		<param name="log_commands" value="corrimudatasb ontime 0.01"/> 
		<param name="log_corrimu_frequency" value="100"/>
		<param name="corr_imu_topic" value="/corr_imu"/>
	</node>
	
	<node name="record_auto_node" pkg="recorder" type="record_all_msgs_node" output= "screen">
		<param name="file_path" value="$(find recorder)/data/state"/>
		<param name="file_name" value="$(arg path_file)_auto.txt"/>
		<param name="imu_topic" value="/corr_imu" />
		<param name="utm_topic" value="/ll2utm_odom" />
		<param name="state_topic" value="/tracking_state"/>
	</node>

</launch>
